### PR TITLE
encode_packed support

### PR DESCRIPTION
**Pull request by krisbitney**
This PR adds the `solidityPack` function, based on the logic from `ethers.js`. It resolves the problems found in `ethers-rs` `encode_packed`.

**Problem with ethers-rs encode_packed**

It seems the behavior of `encode_packed` in ethers-rs is different from the behavior in ethers.js. I'm not sure if this is an issue, but it looks like the Solidity docs agree with ethers-js https://docs.soliditylang.org/en/latest/abi-spec.html#non-standard-packed-mode.

A negative i16 `-1` seems to be interpreted by ethers-rs as a uint256.
```ts
types: [ "int16", "uint48" ]
values: [ "-1", "12" ]
ethers-js: "0xffff00000000000c"
ethers-rs: "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0c"
```

And integers are encoded without padding.
```ts
types: [ "uint256", "uint256" ]
values: [ "8", "16" ]
ethers-js: "0x00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000010"
ethers-rs: "0x0810"
```

### CODE REVIEW

**Comment by krisbitney** [comment]
I created an issue on the ethers-rs github here: https://github.com/gakonst/ethers-rs/issues/2257

**Comment by dOrgJelli** [comment]
Hey @krisbitney, after reading through, I'm thinking we'd be best off re-implement the logic within ethers.js, as it doesn't look too hard to replicate: https://github.com/ethers-io/ethers.js/blob/278f84174409b470fa7992e1f8b5693e6e5d2dac/src.ts/hash/solidity.ts#L83

Thoughts?

File: wraps/utils/tests/e2e.spec.ts [hunk 1]
```diff
@@ -237,6 +239,83 @@ describe("Ethereum Wrapper", () => {
...
+          "address": "0x742d35Cc6634C0532925a3b844Bc454e4438f44e",
+          "bool": "true",
+          "uint8": "255",
+          "uint16": "65535",
+          "uint32": "4294967295",
+          "uint64": "18446744073709551615",
+          "uint128": "340282366920938463463374607431768211455",
+          "uint256": "115792089237316195423570985008687907853269984665640564039457584007913129639935",
+          "uint": "115792089237316195423570985008687907853269984665640564039457584007913129639935",
+          "int8": "127",
```
Comment (Niraj-Kamdar) [inline wraps/utils/tests/e2e.spec.ts:294]: Can you also add tests for negative numbers since you described that it can cause issues?

Comment (krisbitney) [reply wraps/utils/tests/e2e.spec.ts:294]: Yes, 100%. I just added another test that has a test case with a negative number for each int type

**Review by Niraj-Kamdar — APPROVED** [review]

**Review by cbrzn — APPROVED** [review]
thank you for pushing this ser :smile:
